clusterloader/cluster_loader.go

```diff
@@ -93,93 +120,63 @@ var _ = framework.KubeDescribe("Cluster Loader [Feature:ManualPerformance]", fun
+	for _, NS := range allNS {
+		if NS.Name == newNS.Name {
+			return allNS
 		}
-		framework.Failf("No pod tuning found for: %s", podTuning)
 	}
-	return nil
-}
-
-// mkPath returns fully qualfied file location as a string
-func mkPath(file string) string {
-	// Handle an empty filename.
-	if file == "" {
-		framework.Failf("No template file defined!")
-	}
-	// TODO: We should enable passing this as a flag instead of hardcoding.
-	return filepath.Join(os.Getenv("GOPATH"), "src/k8s.io/perf-tests/clusterloader/content/", file)
+	return append(allNS, newNS)
 }
 
 // createTemplate does regex substitution against the template file, then creates the template
-func createTemplate(baseName string, ns *v1.Namespace, configPath string, numObjects int, tuning *clusterloaderframework.TuningSetType) {
+func createTemplate(baseName string, ns *v1.Namespace, configPath string, numObjects int, tuning *clusterloaderframework.TuningSet) error {
 	// Try to read the file
 	content, err := ioutil.ReadFile(configPath)
 	if err != nil {
-		framework.Failf("Error reading file: %s", err)
+		return err
 	}
 
 	// ${IDENTIFER} is what we're replacing in the file
-	regex, err := regexp.Compile("\\${IDENTIFIER}")
-	if err != nil {
-		framework.Failf("Error compiling regex: %v", err)
-	}
+	regex := regexp.MustCompile("\\${IDENTIFIER}")
 
 	for i := 0; i < numObjects; i++ {
 		result := regex.ReplaceAll(content, []byte(strconv.Itoa(i)))
 
 		tmpfile, err := ioutil.TempFile("", "cl")
 		if err != nil {
-			framework.Failf("Error creating new tempfile: %v", err)
+			return err
 		}
 		defer os.Remove(tmpfile.Name())
 
 		if _, err := tmpfile.Write(result); err != nil {
-			framework.Failf("Error writing to tempfile: %v", err)
+			return err
 		}
 		if err := tmpfile.Close(); err != nil {
-			framework.Failf("Error closing tempfile: %v", err)
+			return err
 		}
 
 		framework.RunKubectlOrDie("create", "-f", tmpfile.Name(), getNsCmdFlag(ns))
 		framework.Logf("%d/%d : Created template %s", i+1, numObjects, baseName)
 
 		// If there is a tuning set defined for this template
 		if tuning != nil {
-			if tuning.Templates.RateLimit.Delay != 0 {
-				framework.Logf("Sleeping %d ms between template creation.", tuning.Templates.RateLimit.Delay)
-				time.Sleep(time.Duration(tuning.Templates.RateLimit.Delay) * time.Millisecond)
+			if tuning.Templates.RateLimit.Delay != "" {
+				framework.Logf("Sleeping %v between template creation.", tuning.Templates.RateLimit.Delay)
+				if err := tuning.Templates.Delay(); err != nil {
+					return err
+				}
 			}
 			if tuning.Templates.Stepping.StepSize != 0 && (i+1)%tuning.Templates.Stepping.StepSize == 0 {
-				framework.Logf("We have created %d templates and are now sleeping for %d seconds", i+1, tuning.Templates.Stepping.Pause)
-				time.Sleep(time.Duration(tuning.Templates.Stepping.Pause) * time.Second)
+				framework.Logf("We have created %d templates; sleep for %v", i+1, tuning.Templates.Stepping.Pause)
```

Comment:
Wouldn't work here since the check is to do with the step sizes and not just the presence of the wait time string.